sklearn/preprocessing/tests/test_target_encoder.py

```diff
@@ -709,6 +711,22 @@ def test_pandas_copy_on_write():
     Non-regression test for gh-27879.
     """
     pd = pytest.importorskip("pandas", minversion="2.0")
-    with pd.option_context("mode.copy_on_write", True):
+    # Pandas currently warns that setting copy_on_write will be removed in pandas 4
+    # (and copy-on-write will always be enabled).
+    # see https://github.com/scikit-learn/scikit-learn/issues/32829
+    # TODO: remove this workaround when pandas 4 is our minimum version
+    if parse_version(pd.__version__) >= parse_version("4.0"):
         df = pd.DataFrame({"x": ["a", "b", "b"], "y": [4.0, 5.0, 6.0]})
         TargetEncoder(target_type="continuous").fit(df[["x"]], df["y"])
+    else:
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                "ignore",
+                message="Copy-on-Write can no longer be disabled, "
+                "setting to False has no impact. This option will "
+                "be removed in pandas 4.0.",
+                category=pd.errors.Pandas4Warning,
```

Comment:
I took a look at the class hierarchy of the warning in the nightly build of pandas, and it is a subclass of the usual DeprecationWarning:

```python
>>> pd.errors.Pandas4Warning.__mro__ 
(pandas.errors.Pandas4Warning,
 pandas.errors.PandasDeprecationWarning,
 pandas.errors.PandasChangeWarning,
 DeprecationWarning,
 Warning,
 Exception,
 BaseException,
 object)
```

So filtering those might work, irrespective of the Pandas version.

```suggestion
                category=DeprecationWarning,
```

Reply:
Also: the `message` argument of `filterwarnings` is regular expression. So it's likely necessary to wrap the verbatim expected message as `re.escape(expected_warning_message)`.